### PR TITLE
GSP-109: Redesign Features

### PR DESCRIPTION
**Pull request by Xuanwo**
Signed-off-by: Xuanwo <github@xuanwo.io>

### CODE REVIEW

**Review by xxchan — APPROVED** [review]

File: rfcs/109-redesign-features.md [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+
+- `stat`: SHOULD check `VirtualDir` before set Dir ObjectMode
+- `create`: SHOULD check `VirtualDir` before accept Dir ObjectMode
+- `list`: SHOULD check `VirtualDir` before set Dir ObjectMode
+- `delete`: SHOULD check `VirtualDir` before delete a dir Object.
+- ...
+
+### VirtualObjectMetadata
+
+We will introduce a new feature called `VirtualObjectMetadata` to allow the service to handle object metadata even when it doesn't have native support.
```
Comment (xxchan) [inline rfcs/109-redesign-features.md:79]: Why do we need to add this?

Comment (Xuanwo) [reply rfcs/109-redesign-features.md:79]: Allow fs to use `xattr` to store object metadata like `content-type` and `content-md5`.

Comment (xxchan) [reply rfcs/109-redesign-features.md:79]: I mean how do we do this before this proposal? I'm a little bit confused why it appears here.

Comment (xxchan) [reply rfcs/109-redesign-features.md:79]: Is this refactoring or adding a new feature?

Comment (Xuanwo) [reply rfcs/109-redesign-features.md:79]: > I mean how do we do this before this proposal? I'm a little bit confused why it appears here.

It's used to be added as `VirutalPairWriteContentMd5` / `VirtualPairWriteContentType` but never implemented.

In this proposal, we unify all of them into `VirtualObjectMetadata`.

Comment (xxchan) [reply rfcs/109-redesign-features.md:79]: Got it. Can you add this explanation to the proposal?

**Review by JinnyYi — APPROVED** [review]

**Pull request merged**